packaging/dependency/kvm.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dependency
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/packaging"
+	"github.com/juju/utils/arch"
+)
+
+// KVM returns a dependency instance for installing KVM support.
+func KVM(arch string) packaging.Dependency {
+	return &kvmDependency{arch: arch}
+}
+
+type kvmDependency struct {
+	arch string
+}
+
+// PackageList implements packaging.Dependency.
+func (dep kvmDependency) PackageList(series string) ([]packaging.Package, error) {
+	var pkgList []string
+
+	switch series {
+	case "centos7", "opensuseleap":
+		return nil, errors.NotImplementedf("installing kvm on %s", series)
+	default:
+		if dep.arch == arch.ARM64 {
+			// ARM64 doesn't support legacy BIOS so it requires Extensible Firmware
```

Comment:
I wonder if we should always install efi, it seems relevant for being forward looking. But not something we should change as part of this patch.